contentcuration/contentcuration/view/internal_views.py

```diff
@@ -105,18 +109,20 @@ def api_channel_structure_upload(request):
     """
     Creates a channel based on the structure sent in the request.
     :param request: POST request containing the tree structure of a channel.
-    :return: The channel_id and channel_link of the newly created channel.
+    :return: The channel_id of the newly created channel.
     """
     data = json.loads(request.body)
     try:
+        channel_id = data['channel_id']
         channel_structure = data['channel_structure']
 
-        print(channel_structure)
+        new_channel = create_channel_from_structure(channel_id, channel_structure, request.user)
+
+        if not data.get('stage'):  # If user says to stage rather than submit, skip changing trees at this step
+            activate_channel(new_channel)
 
         return HttpResponse(json.dumps({
             'success': True,
```

Comment:
Could you add `'channel_id': new_channel.pk,` so that the ricecooker can use the newly created channel id?